File: JS/Factorial_Iterative.js
//Program to find the factorial of a number iteratively


//prompts the user and inputs the number 
console.log("This program finds the factorial of a number.\n");
var n = prompt("Which number would you like to input? ");

n = parseInt(n);

var iterator;
var factorial_total;

function fact(n) {
  for (iterator = 1; iterator < n; iterator++) {
    factorial_total = factorial_total * iterator;
  }
}

fact(n)

print("The factorial of ")
print(n)
print(" is ")
print(factorial_total)

File: C++/Factorial_iterative.cpp
#include <iostream>


int factorial(int n) {
    int iterator = 1;
    int factorial_total = 1;
    if (iterator < n)
        iterator = iterator + 1;
    factorial_total = factorial_total * iterator;
    return factorial_total;
}

int main() {

    int n = 0;
    std::cout << "This program finds the factorial of a number.\n" << std::endl;
    std::cout << "Which number would you like to find?\n" << std::endl;
    std::cin >> n;

    std::cout << "The factorial of " << n << " is " << factorial(n);

    return 0;
}



